File: java-web-spring-boot-3/src/main/java/com/vndevteam/javawebspringboot3/application/demo/DemoReqDto.java
package com.vndevteam.javawebspringboot3.application.demo;

import com.fasterxml.jackson.annotation.JsonProperty;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Size;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
public class DemoReqDto {
    @NotBlank(message = "{email.not.blank}")
    @JsonProperty("email")
    private String email;

    @Size(
            min = 5,
            max = 20,
            message = "The text1 '${validatedValue}' must be between {min} and {max} characters long"
    )
    @JsonProperty("text1")
    private String text1;

    @Size(min = 5, max = 20, message = "{text2.length.between.min.max}")
    @JsonProperty("text2")
    private String text2;
}

File: kotlin-webmvc-spring-boot-3/src/main/kotlin/com/vndevteam/kotlinwebmvcspringboot3/application/demo/DemoReqDto.kt
package com.vndevteam.kotlinwebmvcspringboot3.application.demo

import com.fasterxml.jackson.annotation.JsonProperty
import jakarta.validation.constraints.NotBlank
import jakarta.validation.constraints.Size

class DemoReqDto {
    @NotBlank(message = "{email.not.blank}") @JsonProperty("email") var email: String? = null

    @Size(
        min = 5,
        max = 20,
        message = "The text1 '\${validatedValue}' must be between {min} and {max} characters long"
    )
    @JsonProperty("text1")
    var text1: String? = null

    @Size(min = 5, max = 20, message = "{text2.length.between.min.max}")
    @JsonProperty("text2")
    var text2: String? = null
}
